lib/apiservers/portlayer/restapi/handlers/scopes_handlers.go

```diff
@@ -259,17 +293,48 @@ func (handler *ScopesHandlersImpl) ScopesUnbindContainer(params scopes.UnbindCon
 
 func toScopeConfig(scope *network.Scope) *models.ScopeConfig {
 	id := scope.ID()
-	subnet := scope.Subnet().String()
+
+	subnet := ""
+	if !ip.IsUnspecifiedIP(scope.Subnet().IP) {
+		subnet = scope.Subnet().String()
+	}
+
 	gateway := ""
 	if !scope.Gateway().IsUnspecified() {
 		gateway = scope.Gateway().String()
 	}
-	return &models.ScopeConfig{
+
+	sc := &models.ScopeConfig{
 		ID:        &id,
 		Name:      scope.Name(),
 		ScopeType: scope.Type(),
 		IPAM:      scope.IPAM().Pools(),
 		Subnet:    &subnet,
 		Gateway:   &gateway,
 	}
+
+	if len(sc.IPAM) == 0 && len(subnet) != 0 {
+		// use subnet as pool
+		sc.IPAM = []string{subnet}
+	}
+
+	eps := scope.Endpoints()
+	sc.Endpoints = make([]*models.EndpointConfig, len(eps))
+	for i, e := range eps {
+		sc.Endpoints[i] = &models.EndpointConfig{
+			Gateway:   e.Gateway().String(),
+			ID:        e.ID(),
+			Name:      e.Container().Name(),
+			Network:   e.Scope().Name(),
+			Container: e.Container().ID().String(),
+		}
+
+		ip := e.IP()
+		if ip != nil {
```

Comment:
If you move this block above line 324 and assign Address in the creation it's easier to read. addr can be "" by default for the case ip == nil.